block/manager.go

```diff
@@ -240,21 +240,26 @@ func (m *Manager) onNodeHealthStatus(event pubsub.Message) {
 // TODO: move to gossip.go
 // onNewGossippedBlock will take a block and apply it
 func (m *Manager) onNewGossipedBlock(event pubsub.Message) {
-	m.retrieverMutex.Lock() // needed to protect blockCache access
-	m.logger.Debug("Received new block via gossip", "n cachedBlocks", len(m.blockCache))
 	eventData := event.Data().(p2p.GossipedBlock)
 	block := eventData.Block
 	commit := eventData.Commit
 
+	_, found := m.blockCache[block.Header.Height]
```

Comment:
this also needs mutex access
maybe move right above
`m.blockCache[block.Header.Height] =`

Reply:
moved